src/parsing/parsing.js

```diff
@@ -128,7 +130,10 @@
       }
     }
     var parsed = parser(string.substr(0, n));
-    return parsed == undefined ? undefined : [parsed, string.substr(n)];
```

Comment:
lol what, did I write that?

If you want to keep the ternary and still make it less bad, it could be

```javascript
return parsed ? [parsed, string.substr(n)] : null;
```

which is much easier to understand.